packages/runtime/src/helpers/files.ts

```diff
@@ -119,14 +118,17 @@ export const moveStaticPages = async ({
 
   const files: Array<string> = []
   const filesManifest: Record<string, string> = {}
-  const moveFile = async (file) => {
+  const moveFile = async (file: string) => {
+    // Strip the initial 'app' or 'pages' directory from the output path
+    const pathname = file.split('/').slice(1).join('/')
+    // .rsc data files go next to the html file
     const isData = file.endsWith('.json')
-    const source = join(root, file)
-    const targetFile = isData ? join(dataDir, file) : file
+    const source = join(outputDir, file)
+    const targetFile = isData ? join(dataDir, pathname) : pathname
     const targetPath = basePath ? join(basePath, targetFile) : targetFile
 
-    files.push(file)
-    filesManifest[file] = targetPath
+    files.push(pathname)
+    filesManifest[pathname] = targetPath
```

Comment:
Are the `app` and `pages` folders stripped from the pathname because we don't care if they're using the old way (pages) or the new way (app) in terms of generating the files manifest?

Reply:
Oh, good spot. The mainfest should still have the full path, because it needs to intercept the filesystem lookup